packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -258,9 +258,21 @@ const SettingsRoutes = ({
   const location = useLocation();
   const isSync = integration != null ? integration.isSync() : false;
 
+  // TODO: we shouldn't have page load and Glean click events in
+  // both of these route groups, move up higher
   useEffect(() => {
     GleanMetrics.pageLoad();
   }, [location.pathname]);
+  useEffect(() => {
+    const handleClick = (event: MouseEvent) => {
+      GleanMetrics.handleClickEvent(event);
+    };
+    document.addEventListener('click', handleClick);
```

Comment:
While it looks to me that Glean should [add this event listener for us](https://github.com/mozilla/glean.js/blob/eca7d5665be4f2e51ec43628c4427c9def35980b/glean/src/core/glean.ts#L338), I guess our `GleanMetrics` implementation is more customized, so recreating this similar to us recreating `pageLoad` above was needed.

Reply:
I debugged into this, and the reason and Glean isn't adding it is that the `uploadEnabled` flag is false. This could be turned on by passing in `uploadEnabled:true` [here](https://github.com/mozilla/fxa/blob/7423300a5121fa72fb267a19a45c4ba5de584382/packages/fxa-settings/src/components/App/index.tsx#L169). I'm not sure if doing so would have other side effects. If not, then setting this flag seems better.

If we do end up using this addEventListener, can we put a comment here and on the GleanMetrics.initialize call? This might help down road if uploadEnabled is ever turned on. _(Perhaps binding twice could result in double logging of events... not sure.)_

Here's a screenshot shot of the debugger state:

![image](https://github.com/user-attachments/assets/cb7d3e73-8052-424c-b128-9fb8904168d2)

Reply:
Oooh thanks for catching. I shouldn't have just assumed this was enabled already so I'll take a closer look!